### PR TITLE
implemented &operator+= function in matrix library

### PR DESCRIPTION
**Pull request by spencersharp2010**
attempted implementation of operator+= function in matrix library. probably could use some additional testing, but seems to work for my test cases

### CODE REVIEW

File: 04_12_2019/development/main.cpp [hunk 1]
```diff
@@ -5,6 +5,40 @@
...
+	for (int i = 0; i < m.numberOfRows(); ++i)
+	{
+		for (int j = 0; j < m.numberOfCols(); ++j)
+		{
+			std::cout << m(i, j) << "\t";
+		}
+		std::cout << std::endl;
+	}
+
+	std::cout << std::endl;
```
Comment (sariug) [inline 04_12_2019/development/main.cpp]: There is the overloading for `<<` operator.  `std::cout<<m;` should be more than enough. Same for the second one.

File: 04_12_2019/development/main.cpp [hunk 2]
```diff
@@ -5,6 +5,40 @@
...
+	
+	// perform addition to itself
+	m.operator+=(m);
```
Comment (sariug) [inline 04_12_2019/development/main.cpp]: it is the worst practice to call operator as the function. What shall it look like is:
`int i = 3; // i is 3`
`i += i;     // i is 6`

File: Matrix/inc/Matrix.hpp [hunk 3]
```diff
@@ -30,6 +30,7 @@ class Matrix
     double &operator()(int x, int y);
     friend bool operator==(const Matrix &lhs, const Matrix &rhs);
     Matrix &operator+(Matrix &mat_1);
+	Matrix &operator+=(const Matrix& mat1);
```
Comment (sariug) [inline Matrix/inc/Matrix.hpp]: please correct the identation

File: Matrix/src/Matrix.cpp [hunk 4]
```diff
@@ -105,6 +105,23 @@ Matrix &Matrix::operator+(Matrix &mat_1)
...
+Matrix &Matrix::operator+=(const Matrix& RHS)
+{
+	size_t nRows = numberOfRows();
+	size_t nColumns = numberOfCols();
+
+	for (int i = 0; i < nRows; ++i)
+	{
+		for (int j = 0; j < nColumns; ++j)
+		{
+			(*this)(i, j) += RHS(i, j);
```
Comment (sariug) [inline Matrix/src/Matrix.cpp]: The operators has right to reach to the private items. Therefore you can directly update the main matrix:
`m_matrix[i][j] += RHS(i, j); ` (@JusticeBoi )

File: Matrix/src/Matrix.cpp [hunk 5]
```diff
@@ -105,6 +105,23 @@ Matrix &Matrix::operator+(Matrix &mat_1)
 	return mat_1;
 }
 
+Matrix &Matrix::operator+=(const Matrix& RHS)
+{
```
Comment (sariug) [inline Matrix/src/Matrix.cpp:109]: What happens if matrices have different sizes? Lets say:

    Matrix A(3,3);
    Matrix B(4,4);
    A+=B;

Would this work ?

**Review by sariug — CHANGES_REQUESTED** [review]
Nice written code. Please update it :) Would be nice to add a few tests more. (using something minus, something plus, something with zero etc..)

File: Matrix/src/Matrix.cpp (hunk 4 above)
Comment (JusticeBoi) [reply Matrix/src/Matrix.cpp]: I would guess they end up producing the same assembly, so it is a matter of style. Perhaps @sariug's suggestion looks a bit cleaner.

**Comment by spencersharp2010** [comment]
Thanks for your comments. I have updated the code with improved syntax as you suggested above, and I added exception handling (try/catch block) to account for mismatched array dimensions

File: Matrix/inc/Matrix.hpp (hunk 3 above)
Comment (spencersharp2010) [reply Matrix/inc/Matrix.hpp]: I checked my second commit, and this still appears to be indented with the github viewer. On my machine, it looks correct though

**Comment by sariug** [comment]
Thank you for the effort. Looks much nicer ! 

Just 3 things : 
1.  Could you mark the resolved conversations from the first review ? 
2. I am actually surprised that the functions are working. The signitures are different ? Could you fix it ?
image
3. Could you fix the indentations in general ?
image

**Review by sariug — APPROVED** [review]

**Pull request merged**